packages/components/src/modifiers/hds-code-editor.ts

```diff
@@ -195,15 +202,12 @@ export default class HdsCodeEditorModifier extends Modifier<HdsCodeEditorSignatu
     positional: PositionalArgs<HdsCodeEditorSignature>,
     named: NamedArgs<HdsCodeEditorSignature>
   ): void {
-    const { hasLineWrapping = false } = named;
+    const { value, hasLineWrapping } = named;
 
-    // if the editor already exists, update the line wrapping
+    // if the editor already exists, update its state
     if (this.editor) {
-      this.editor.dispatch({
-        effects: this.lineWrappingCompartment.reconfigure(
-          hasLineWrapping ? EditorView.lineWrapping : []
-        ),
-      });
+      // eslint-disable-next-line ember/no-runloop
+      next(this, this._updateEditorState, { value, hasLineWrapping });
```

Comment:
I don't think the runloop should be necessary here - you should only set `value` once. The issue you were seeing is from using the same function with the external button and the `onInput` function below (I'll comment down there to make it clearer).